Misc/NEWS.d/next/Library/2023-11-27-19-54-43.gh-issue-59013.chpQ0e.rst

```diff
@@ -0,0 +1 @@
+Set breakpoint on the first executable line in :mod:`pdb` instead of function definition when using ``break func``
```

Comment:
I don't think this is clear. "The first executable line in pdb" sounds like the first executable line in the pdb module. Also, what does "break func" mean here, and what is the problem with it?

Reply:
I don;t actually understand what this PR is doing. When do we need to break on uncompiled source code?

Reply:
Let me try to explain the purpose of this PR before addressing other comments.

If you have a code like this:
```python
breakpoint()

def f():
    # comment
    pass

f()
```

When you launch `pdb`, `f` does not exist. The source code is compiled, but `MAKE_FUNCTION` has not executed. Now when you do `break f` (or `b f`) in pdb, it will set a breakpoint on f, but that's entirely based on regular expression and pdb will show the breakpoint is at line 3 - the line of `def f():`. However, that is not where the actual breakpoint is. If you continue to execute the code, you'll stop at line 5 - `pass`. That's confusing to the users.

The purpose of this PR, is to compile a small piece of the function code after the regular expression match, to find the first executable line of the function - so when the user `break` on a function that does not exist yet, it can find the correct line number. In this case, `break f` will result in a breakpoint (showing that it's) on line 5 (the mechanism is actually slightly more complicated).

Reply:
Thanks, this makes sense.

Reply:
Is it better now?

Reply:
yes